### PR TITLE
AlarmConfiguration#autoscalingGroupName is not being used

### PR DESCRIPTION
**Pull request by fabiokung**
Remove dead code.

### CODE REVIEW

File: titus-api/src/main/java/com/netflix/titus/api/appscale/store/mixin/AlarmConfigurationMixIn.java [hunk 1]
```diff
@@ -31,7 +31,6 @@
             @JsonProperty("region") String region,
             @JsonProperty("actionsEnabled") Optional<Boolean> actionsEnabled,
             @JsonProperty("comparisonOperator") ComparisonOperator comparisonOperator,
-            @JsonProperty("autoScalingGroupName") String autoScalingGroupName,
```
Comment (tbak) [inline titus-api/src/main/java/com/netflix/titus/api/appscale/store/mixin/AlarmConfigurationMixIn.java]: Removing this attribute may prevent loading of the existing records from the database. We need to provide a hint to  Jackson deserializer to ignore this value if present.

Comment (corindwyer) [reply titus-api/src/main/java/com/netflix/titus/api/appscale/store/mixin/AlarmConfigurationMixIn.java]: I believe it is the `@JsonIgnoreProperties(ignoreUnknown = true)` annotation.

Comment (fabiokung) [reply titus-api/src/main/java/com/netflix/titus/api/appscale/store/mixin/AlarmConfigurationMixIn.java]: good catch, I thought this was always the default

Comment (fabiokung) [reply titus-api/src/main/java/com/netflix/titus/api/appscale/store/mixin/AlarmConfigurationMixIn.java]: Done. PTAL

**Pull request merged**